lib/seattleflu/id3c/cli/command/etl/redcap_det_scan.py

```diff
@@ -0,0 +1,534 @@
+    """ Returns a FHIR Condition resource. """
+    mapped_symptom_name = map_symptom(symptom_name)
+    if not mapped_symptom_name:
+        return None
+
+    # XXX TODO: Define this as a TypedDict when we upgrade from Python 3.6 to
+    # 3.8.  Until then, there's no reasonable way to type this data structure
+    # better than Any.
+    #   -trs, 24 Oct 2019
+    condition: Any = {
+        "resourceType": "Condition",
+        "id": mapped_symptom_name,
+        "code": {
+            "coding": [
+                {
+                    "system": f"{INTERNAL_SYSTEM}/symptom",
+                    "code": mapped_symptom_name
+                }
+            ]
+        },
+        "onsetDateTime": record["symptom_duration"], # YYYY-MM-DD in REDCap
+        "subject": patient_reference
+    }
+
+    return condition
+
+
+def create_specimen(record: dict, patient_reference: dict) -> tuple:
+    """ Returns a FHIR Specimen resource entry and reference """
+    def specimen_barcode(record: Any) -> str:
+        """
+        Given a REDCap *record*, returns the barcode or corrected barcode if it
+        exists.
+        """
+        barcode = record['return_utm_barcode'] or record['pre_scan_barcode']
+
+        if not barcode:
+            barcode = record['utm_tube_barcode_2']
+            reentered_barcode = record['reenter_barcode']
+
+            if record['barcode_confirm'] == "No":
+                #TODO: Figure out why 'corrected_barcode' doesn't always exist?
+                barcode = record.get('corrected_barcode')
+
+        return barcode
+
+    barcode = specimen_barcode(record)
+    if not barcode:
+        LOG.warning("Could not create Specimen Resource due to lack of barcode.")
+        return None, None
+
+    specimen_identifier = create_identifier(
+        system = f"{INTERNAL_SYSTEM}/sample",
+        value = barcode
+    )
+
+    # YYYY-MM-DD in REDCap
+    collected_time = record['collection_date'] or None
+
+    # YYYY-MM-DD HH:MM:SS in REDCap
+    received_time = record['samp_process_date'].split()[0] if record['samp_process_date'] else None
+
+    specimen_type = 'NSECR'  # Nasal swab.  TODO we may want shared mapping function
+    specimen_resource = create_specimen_resource(
+        [specimen_identifier], patient_reference, specimen_type, received_time, collected_time
+    )
+
+    return create_entry_and_reference(specimen_resource, "Specimen")
+
+
+def create_questionnaire_response(record: dict, patient_reference: dict,
+    encounter_reference: dict) -> Optional[dict]:
+    """ Returns a FHIR Questionnaire Response resource entry. """
+
+    def combine_checkbox_answers(coded_question: str) -> Optional[List]:
+        """
+        Handles the combining "select all that apply"-type checkbox
+        responses into one list.
+
+        Uses our in-house mapping for race.
+        """
+        regex = rf'{coded_question}___[\w]*$'
```

Comment:
I think in our specific current usages it's not an issue, but generally-speaking it is best to be defensive and [escape](https://docs.python.org/3/library/re.html#re.escape) interpolations into regex patterns, e.g.:

```python
regex = rf'{re.escape(coded_question)}___[\w]*$'
```

This prevents metachars in the interpolation from being misinterpreted.